src/main/java/com/google/code/or/OpenReplicator.java

```diff
@@ -286,6 +289,11 @@ protected void setupHeartbeatPeriod() throws Exception {
 		query.getFirst("SET @master_heartbeat_period = " + nanoSeconds);
 	}
 
+	protected void setupSlaveUUID() throws Exception {
+		final Query query = new Query(this.transport);
+		query.getFirst("SET @slave_uuid = '" + this.uuid + "'");

```

Comment:
afaict the value of `slave_uuid` is either `server_uuid` or autogenerated.  But the value that's passed to the mysql server is `slave_uuid`: https://github.com/mysql/mysql-server/blob/5.7/sql/rpl_slave.cc#L2433